### PR TITLE
Add sfo gateway configuration

### PR DESCRIPTION
**Pull request by pablothedude**
The configuration parameters need to be added in order to be able to configure SFO in EB.

I did not implement the DI of the parameters so that the best option could be determined when implementing the actual SFO callout.

https://www.pivotaltracker.com/story/show/166702276

### CODE REVIEW

File: tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoGatewayLoaMappingTest.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
```
Comment (MKodde) [inline tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoGatewayLoaMappingTest.php]: 2019

File: tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoGatewayLoaMappingTest.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\EngineBlockBundle\Tests;
+
+use OpenConext\EngineBlockBundle\Sfo\SfoGatewayLoaMapping;
+use PHPUnit_Framework_TestCase as TestCase;
+
+class SfoGatewayLoaMappingTest extends TestCase
```
Comment (MKodde) [inline tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoGatewayLoaMappingTest.php]: Nice mapping helper object! Could you please add some sad-flow test coverage? 100% coverage FTW

File: application/configs/application.ini [hunk 3]
```diff
@@ -161,6 +161,16 @@ pdp.username = "pdp_admin"
...
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;; SFO SETTINGS ;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+sfo.gateway.loa.mapping[http://test.surfconext.nl/assurance/loa2] = "https://gateway.tld/authentication/loa2"
+sfo.gateway.loa.mapping[http://test.surfconext.nl/assurance/loa3] = "https://gateway.tld/authentication/loa3"
+sfo.gateway.loa.loa1 = "https://gateway.tld/authentication/loa1"
+sfo.gateway.sso.entityId = "https://engine.surfconext.nl/authentication/sp/sso"
+sfo.gateway.sso.ssoLocation = "https://gateway.vm.openconext.org/sfo/sso"
+sfo.gateway.sso.keyFile = "/etc/openconext/engineblock.pem"
```
Comment (MKodde) [inline application/configs/application.ini]: Aligned nicely with the other config settings! :+1:

File: tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoEndpointTest.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\EngineBlockBundle\Tests;
+
+use OpenConext\EngineBlockBundle\Sfo\SfoEndpoint;
+use PHPUnit_Framework_TestCase as TestCase;
+
+class SfoEndpointTest extends TestCase
```
Comment (MKodde) [inline tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoEndpointTest.php]: Again, please add non-happy flow code coverage. And update the copyright year to 2019 please!

File: tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoEndpointTest.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * @test
+     * @group Sfo
+     */
+    public function the_sfo_endpoint_object_should_be_successful_populated()
+    {
+        $endpoint = new SfoEndpoint('entity-id', 'https://sso-location', '/etc/openconext/engineblock.crt');
```
Comment (MKodde) [inline tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoEndpointTest.php]: To make the test safer, I suggest adding a dummy key to the test resources. Or reuse one found in `OpenConext-engineblock/tests/resources`

File: bin/composer/dump-required-ini-params.sh [hunk 6]
```diff
@@ -236,7 +236,14 @@ $ymlContent = array(
                 'attributeDefinitionFile', $projectRoot . 'application/configs/attributes-v2.2.0.json'
             )
         ),
-        'monitor_database_health_check_query'                     => $config->get('openconext.monitor_bundle_health_query', '')
+        'monitor_database_health_check_query'                     => $config->get('openconext.monitor_bundle_health_query', ''),
+
+        // SFO
+        'sfo.gateway.loa.mapping'                                 => $config->get('sfo.gateway.loa.mapping', array())->toArray(),
+        'sfo.gateway.loa.loa1'                                    => escapeYamlValue($config->get('sfo.gateway.loa.loa1')),
```
Comment (MKodde) [inline bin/composer/dump-required-ini-params.sh]: It does not harm, but escaping the YAML values does not add any value in these config settings as far as I can see.

File: application/configs/application.ini (hunk 3 above)
Comment (MKodde) [reply application/configs/application.ini]: As these are optional settings, I suggest commenting them out by default.

File: tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoGatewayLoaMappingTest.php (hunk 1 above)
Comment (pablothedude) [reply tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoGatewayLoaMappingTest.php]: done

File: tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoEndpointTest.php (hunk 4 above)
Comment (pablothedude) [reply tests/unit/OpenConext/EngineBlockBundle/Sfo/SfoEndpointTest.php]: Done

File: bin/composer/dump-required-ini-params.sh (hunk 6 above)
Comment (pablothedude) [reply bin/composer/dump-required-ini-params.sh]: I'll leave it in.

**Pull request merged**